targets/curie_bsp/setup.py

```diff
@@ -159,62 +170,83 @@ def create_makefile_in_curie(curie_path):
+    jerry_objects = build_obj_y(jerry_data['sources'])
+    jerry_defines = jerry_data['cflags']
+    jerry_build_dirs = build_mkdir(jerry_data['dirs'])
+
+    quark_include_paths = [
+        'include',
+        'jerryscript',
+        'jerryscript/jerry-libm/include',
+        'jerryscript/targets/curie_bsp/include'
+    ] + list(jerry_data['dirs'])
+
+    quark_includes = [
+        '-Wno-error',
+    ] + ['-I$(PROJECT_PATH)/quark/%s' % dir for dir in quark_include_paths]
+
+    quark_cflags = build_cflags_y(jerry_defines + quark_includes)
+
+    quark_kbuild_content = '''
+{cflags}
+
 obj-y += main.o
-subdir-cflags-y += -Wno-error
-subdir-cflags-y += -I$(PROJECT_PATH)/quark/include
-subdir-cflags-y += -I$(PROJECT_PATH)/quark/jerryscript
-subdir-cflags-y += -I$(PROJECT_PATH)/quark/jerryscript/targets/curie_bsp/include
-'''
-    jerry_core_path = os.path.join(curie_path, 'wearable_device_sw/projects/' \
-                                                + project_name \
-                                                + '/quark/jerryscript/jerry-core/')
-    for line in os.walk(jerry_core_path):
-        npos = line[0].find(project_name)
-        out_put += 'subdir-cflags-y += -I$(PROJECT_PATH)' \
-                + line[0][npos + len(project_name):] + '\n'
+{objects}
 
+build_dirs:
+{dirs}
 
-    jerry_libm_path = os.path.join(curie_path, 'wearable_device_sw/projects/' \
-                                                + project_name \
-                                                + '/quark/jerryscript/jerry-libm/')
+$(OUT_SRC): build_dirs
+'''.format(objects=jerry_objects, cflags=quark_cflags, dirs=jerry_build_dirs)
 
-    for line in os.walk(jerry_libm_path):
-        npos = line[0].find(project_name)
-        out_put += 'subdir-cflags-y += -I$(PROJECT_PATH)' \
-                + line[0][npos + len(project_name):] + '\n'
+    write_file(quark_kbuild_path, quark_kbuild_content)
 
-    file_to_be_created = open(file_path, 'w+')
-    file_to_be_created.write(out_put)
-    file_to_be_created.close()
 
-def main(curie_path, jerry_path):
-    build_soft_links(curie_path, jerry_path)
-    create_kbuild_in_jerryscript(jerry_path)
-    create_makefile_in_curie(curie_path)
+def main(curie_path, project_name, jerry_path):
+    project_path = os.path.join(curie_path, 'wearable_device_sw', 'projects', project_name)
 
+    build_soft_links(project_path, jerry_path)
+
+    create_root_kbuild(project_path)
+    create_root_makefile(project_path)
+    create_arc_kbuild(project_path)
+    create_quark_kbuild(project_path, jerry_path)
 
 
 if __name__ == '__main__':
-    if len(sys.argv) != 2:
-        print 'Usage:'
-        print sys.argv[0] + ' [full or relative path of Curie_BSP]'
-    else:
-        file_dir = os.path.dirname(os.path.abspath(__file__))
-        jerry_path = os.path.join(file_dir, "..", "..")
-        curie_path = os.path.join(os.getcwd(), sys.argv[1])
-        main(curie_path, jerry_path)
+    import sys
+
+    if len(sys.argv) < 2:
```

Comment:
Just wanted to be a bit more fault tolerant (accidentally putting two paths there :)) but you are right it would be better to leave it as `!=`